command.go

```diff
@@ -257,7 +257,7 @@ func (cmd *Command) setupDefaults(osArgs []string) {
 	}
 
 	if cmd.EnableShellCompletion || cmd.Root().shellCompletion {
-		completionCommand := buildCompletionCommand()
+		completionCommand := buildCompletionCommand(cmd.Name)
```

Comment:
```suggestion
		completionCommand := buildCompletionCommand(osArgs[0])
```

That way you dont have to worry about binary name and cmd.Name ?

Reply:
Thanks. Is `os.Args[0]` always the binary name though? Will it wokr when the binary is called with a shell alias, not directly?

I'll have to check...